include/alpaka/queue/QueueCudaRtSync.hpp

```diff
@@ -247,19 +269,37 @@ namespace alpaka
                         pCallbackSynchronizationData.get(),
                         0u));
 
-                    // If the callback has not yet been called, we wait for it.
-                    std::unique_lock<std::mutex> lock(pCallbackSynchronizationData->m_mutex);
-                    if(!pCallbackSynchronizationData->notified)
-                    {
-                        pCallbackSynchronizationData->m_event.wait(
-                            lock,
-                            [pCallbackSynchronizationData](){
-                                return pCallbackSynchronizationData->notified;
+                    // We start a new std::thread which stores the task to be executed.
```

Comment:
Do we always enqueue CUDA callbacks with alpaka?

Reply:
OK after reviewing the corresponding parts again I found that this is the code to create a callback introduced in #373. 
The code itself looks like it is the kernel enqueue stuff but kernel start in the name space alpaka::exec.

Reply:
I have not changed anything about the `cudaStreamAddCallback` here. The bug I found was that the callback was directly executed within the CUDA callback thread. In the CUDA callback thread you are not allowed to do CUDA calls, which I did in my test. This had already been fixed in the Async version. I simply copied the fix from over there.